internal/dag/policy.go

```diff
@@ -123,25 +124,32 @@ func escapeHeaderValue(value string) string {
 }
 
 // ingressRetryPolicy builds a RetryPolicy from ingress annotations.
-func ingressRetryPolicy(ingress *v1beta1.Ingress) *RetryPolicy {
+func ingressRetryPolicy(ingress *v1beta1.Ingress, log logrus.FieldLogger) *RetryPolicy {
```

Comment:
Feels like this is wiring the logger way down into the code, but I don't see any other way to do it.

Reply:
Yeah, I didn't love doing this either. The alternative I considered was having this function return a partial policy along with an error and letting the caller log the error, but this pattern seemed more obvious to me.

Reply:
I guess we could make these methods on the ingress processor, and then they could access the logger as a field of the processor. 🤷‍♂️